components/centraldashboard/public/components/landing-page.js

```diff
@@ -31,10 +31,108 @@ import utilitiesMixin from './utilities-mixin.js';
 export class LandingPage extends mixinBehaviors([AppLocalizeBehavior], utilitiesMixin(PolymerElement)) {
     static get template() {
         const vars = {logo};
-
         return html([
             `<style>${css.toString()}</style>${template(vars)}`]);
     }
+
+    static get properties() {
+        return {
+            userDetails: {type: Object, observer: '_onUserDetails'},
+            namespaceName: String,
+            emailAddress: String,
+            error: Object,
+            flowComplete: {type: Boolean, value: false},
+            waitForRedirect: {type: Boolean, value: false},
+            showAPIText: {type: Boolean, value: false},
+        };
+    }
+
+    ready() {
+        super.ready();
+    }
+
+    _onUserDetails(d) {
+        this.emailAddress = this.userDetails;
+        this.generateNamespace(this.userDetails);
+    }
+
+    async generateNamespace(email) {
+        // Since email includes an @ , we split to the left side of it'
+        const name = email.split('@', 1);
+        let ns = name[0];
+        ns = ns
+            .replace(/[^\w]|\./g, '-')
+            .replace(/^-+|-+$|_/g, '')
+            .toLowerCase();
+
+        this.getNamespaces(ns);
+    }
+
+    async getNamespaces(ns) {
+        await fetch(
+            `/api/namespaces/`, {
+                method: 'GET',
+                headers: {
+                    'Content-Type': 'application/json',
+                },
+            }
+        )
+            .then((res) => res.json())
+            .then((data) => {
+                const namespaceNames = [];
+                data.forEach((element) => {
+                    namespaceNames.push(element.metadata.name);
+                });
+                let counter = 0;
+                while (namespaceNames.includes(ns)) {
+                    counter++;
+                    ns = ns + counter;
```

Comment:
Since ns is a string, wont this just keep appending the counter to the value? So "example-ns", "example-ns1", "example-ns12", "example-ns123", and so on?

Reply:
oops yes